coursera/api.py

```diff
@@ -777,10 +792,17 @@ def extract_links_from_supplement(self, element_id):
         """
         logging.debug('Gathering supplement URLs for element_id <%s>.', element_id)
 
-        dom = get_page(self._session, OPENCOURSE_SUPPLEMENT_URL,
-                       json=True,
-                       course_id=self._course_id,
-                       element_id=element_id)
+        try:
+            dom = get_page(self._session, OPENCOURSE_SUPPLEMENT_URL,
+                           json=True,
+                           course_id=self._course_id,
+                           element_id=element_id)
+        except HTTPError as ex:
+            FORBIDDEN = 403
+            if ex.response.status_code == FORBIDDEN:
+                return {}
```

Comment:
This is bad practice to return `no data` result upon an error in this case. If error happens, report it. There are other options to consider: 1. return`None` 2. reraise HTTPError 3. catch HTTPError and reraise something else 4. Do not catch HTTP error here. The actual choice depends on how this method is being used up the call tree.